### ISSUE TITLE
Dividing data into train/test based on Time

### ISSUE THREAD

**Issue by Richie-Peak** [lyst/lightfm#657]
Firstly, thank you so much for making a great package!

Currently LightFM cannot divide data into train/test temporally, i.e. "train on all data except last 6 months, test on data from last 6 months". If I wrote this, would it simply be a case of:

- Test set is a sparse matrix of all the data
- Training set is a sparse matrix of all the data as of time _t_, with all data after _t_ set to _x_ where _x_ is whatever a non interaction is. So if my data is 1 for "customer bought product" and 0 for "customer has never bought product" then set data > _t_ as 0.

Then it would fit into LightFM's `precision_at_k` and `auc_score()` functions, as you could pass train and test to `train_interactions` and `test_interactions` respectively?

**Comment by EdwardALockhart** [comment]
I think I have a solution for you. Look at my code here.

https://github.com/lyst/lightfm/issues/663
